AutoCollection/ServerRequestParser.ts

```diff
@@ -169,6 +169,29 @@ class ServerRequestParser extends RequestParser {
         return value;
     }
 
+    private parseHeaders(request:http.ServerRequest, parentId?: string) {
+        this.rawHeaders = request.headers || (<any>request).rawHeaders;
+        this.userAgent = request.headers && request.headers["user-agent"];
+        this.sourceCorrelationId = Util.getCorrelationContextTarget(request, RequestResponseHeaders.requestContextSourceKey);
+
+        if (request.headers) {
+            this.correlationContext = request.headers[RequestResponseHeaders.correlationContextHeader];
+
+            if (request.headers[RequestResponseHeaders.requestIdHeader]) {
+                this.parentId = request.headers[RequestResponseHeaders.requestIdHeader];
+                this.requestId = CorrelationIdManager.generateRequestId(this.parentId);
```

Comment:
I'm not sure that I understand the purpose of the code-supplied parentId. From what I understand of the spec, if an incoming request specifies a `Request-Id`, then this request is part of a larger tracked operation, and we should take that Id and extend it appropriately, which is what I do here. What I don't understand is how to combine an incoming `Request-Id` with an ambient/passed-in `Request-Id`. If the incoming request does not specify a `Request-Id` then we will go into the `else` case which checks for legacy headers, and if they are not found it will use the passed-in parentId.